server/services/notify/notifymentions/extract.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package notifymentions
+
+import "strings"
+
+const (
+	defPrefixLines    = 2
+	defPrefixMaxChars = 100
+	defSuffixLines    = 2
+	defSuffixMaxChars = 100
+)
+
+type limits struct {
+	prefixLines    int
+	prefixMaxChars int
+	suffixLines    int
+	suffixMaxChars int
+}
+
+func newLimits() limits {
+	return limits{
+		prefixLines:    defPrefixLines,
+		prefixMaxChars: defPrefixMaxChars,
+		suffixLines:    defSuffixLines,
+		suffixMaxChars: defSuffixMaxChars,
+	}
+}
+
+// extractText returns all or a subset of the input string, such that
+// no more than `prefixLines` lines preceding the mention and `suffixLines`
+// lines after the mention are returned, and no more than approx
+// prefixMaxChars+suffixMaxChars are returned.
+func extractText(s string, mention string, limits limits) string {
+	if !strings.HasPrefix(mention, "@") {
+		mention = "@" + mention
+	}
+	lines := strings.Split(s, "\n")
+
+	// find first line with mention
+	var found int
+	for i, l := range lines {
+		if strings.Contains(l, mention) {
+			found = i
+			break
+		}
+	}
```

Comment:
Should we simply return "", if not found.

Reply:
Done via https://github.com/mattermost/focalboard/pull/1147/commits/1c131c9f301a8e2cc9e5258516de9473d0533e73